src/Data/Vec/Properties.agda

```diff
@@ -1002,6 +999,29 @@ zipWith-replicate₂ _⊕_ []       y = refl
 zipWith-replicate₂ _⊕_ (x ∷ xs) y =
   cong (x ⊕ y ∷_) (zipWith-replicate₂ _⊕_ xs y)
 
+------------------------------------------------------------------------
+-- iterate
+
+length-iterate : ∀ n f (x : A) → length (iterate f x {n}) ≡ n
```

Comment:
No case analysis required; actually this lemma is not required... its proof is `refl` *definitionally* by virtue of the definition of `length` for vectors. Consider omitting.